compiler/rustc_parse/src/lexer/tokentrees.rs

```diff
@@ -106,7 +109,20 @@ impl<'a> TokenTreesReader<'a> {
         // Parse the token trees within the delimiters.
         // We stop at any delimiter so we can try to recover if the user
         // uses an incorrect delimiter.
-        let tts = self.parse_token_trees(/* is_delimited */ true)?;
+        let (tts, res) = self.parse_token_trees(/* is_delimited */ true);
+        if let Err(mut err) = res {
+            // If there are unclosed delims, see if there are diff markers and if so, point them
+            // out instead of complaining about the unclosed delims.
+            let mut parser = crate::stream_to_parser(self.string_reader.sess, tts, None);
+            while parser.token != token::Eof {
+                if let Err(mut diff_err) = parser.err_diff_marker() {
+                    diff_err.emit();
+                    err.delay_as_bug();
```

Comment:
As I mentioned in the neighbor comment https://github.com/rust-lang/rust/pull/116712#discussion_r1374432093 the upper level logic here expects errors to be buffered, and not emitted immediately, but here the error is emitted immediately.

I also don't understand the purpose of `delay_as_bug` here.

Reply:
> I also don't understand the purpose of delay_as_bug here.

It is so that only the diff errors are emitted, silencing the unclosed delim errors because they would be unnecessary noise.

I'll address the rest of the comment.

Reply:
Changed to bubble the errors as a Vec.